### PR TITLE
Recall msg

### PR DESCRIPTION
**Pull request by SilentQianyi**
## Issues

https://github.com/wechaty/wechaty/issues/1885

### CODE REVIEW

File: src/puppet.ts [hunk 1]
```diff
@@ -760,6 +760,7 @@ export abstract class Puppet extends EventEmitter {
   public abstract async messageSendFile (receiver: Receiver, file: FileBox)                          : Promise<void | string>
   public abstract async messageSendUrl (receiver: Receiver, urlLinkPayload: UrlLinkPayload)          : Promise<void | string>
   public abstract async messageSendMiniProgram (receiver: Receiver, miniProgramPayload: MiniProgramPayload)          : Promise<void | string>
+  public abstract async messageRecall (receiver: Receiver, svrMsgId: string) : Promise<boolean>
```
Comment (huan) [inline src/puppet.ts]: 1. What's the reason that we need the `Receiver` as the argument?
1. You need to follow the naming style of other code. `svrMsgId` should be `messageId`.

**Comment by SilentQianyi** [comment]
> Please describe your design in detail in the related issue.
> 
> Thank you very much.

消息撤回方法messageRecall

File: src/puppet.ts (hunk 1 above)
Comment (SilentQianyi) [reply src/puppet.ts]: 在底层实现中所需参数

> {
>&nbsp;&nbsp; fromUserName: string, 
>&nbsp;&nbsp; msgId: string,
>&nbsp;&nbsp; toUserName: string
> }

因为三个参数都必须拥有，其中toUserName是接收方id，所以需要receiver参数。

Comment (windmemory) [reply src/puppet.ts]: I think we should retrieve these information (`fromUserName`, `toUserName`) from puppet instance.

Comment (SilentQianyi) [reply src/puppet.ts]: yes. I fix it

**Pull request merged**